conda_libmamba_solver/index.py

```diff
@@ -43,10 +43,27 @@ def __init__(
         self._repos = []
         self._pool = api.Pool()
 
+        installed_records = list(installed_records)  # in case we are passed a lazy iterator
         installed_repo = self._repo_from_records(self._pool, "installed", installed_records)
         installed_repo.set_installed()
         self._repos.append(installed_repo)
 
+        # See https://github.com/conda/conda/issues/11790
+        channels_from_installed = []
+        for record in installed_records:
+            if record.channel.auth or record.channel.token:
+                # skip if the channel has authentication info, because
+                # it might cause issues with expired tokens and what not
+                continue
+            if record.channel.name in ("@", "<develop>", "pypi"):
```

Comment:
Do you know if we have a definite list of these pseudo channel names in conda that we don't have to hard-code it here?

Reply:
I checked but I couldn't find it. Looks like they were added as needed for edge cases found during the development of orthogonal features. `@` is for virtual channels, `<develop>` seems to come from `pip install -e` and `pypi` is found for `pip install` that do not have a `conda-meta` entry, as far as I can understand.